src/main/scala-2.12/org/ergoplatform/appkit/ergotool/dex/CancelOrderCmd.scala

```diff
@@ -109,12 +109,20 @@ object CancelOrder {
   }
 
   // TODO: extract to use in other commands
-  case class OutBoxProto(getValue: Long, tokens: Seq[ErgoToken], contract: ErgoContract) {
+  case class OutBoxProto(getValue: Long, tokens: Seq[ErgoToken], registers: Seq[ErgoValue[_]], contract: ErgoContract) {
 
     def toOutBox(builder: OutBoxBuilder): OutBox = {
-      val builder2 = builder
-        .value(getValue)
-        .contract(contract)
+      val builder2 =
+        if (registers.nonEmpty) {
+          builder
+            .value(getValue)
+            .contract(contract)
+            .registers(registers: _*)
+        } else {
+          builder
+            .value(getValue)
+            .contract(contract)
+        }
```

Comment:
the .registers method should handle empty sequence correctly, this `if` shouldn't be necessary

Reply:
Nope - `java.lang.IllegalArgumentException: At least one register should be specified`. See http://github.com/aslesarenko/ergo-appkit/blob/a18d3ad87500259d3ef42bfbe58e807a3d4c9f11/lib-impl/src/main/java/org/ergoplatform/appkit/impl/OutBoxBuilderImpl.java#L49-L49